File: C/Bubble_Sort.c
#include <stdio.h>

// declare functions implemented after main
void bubbleSort(int *arr, int n);
void printArr(int *arr, int n);

int main()
{
	// initialize array and get length
	int arr[] = { 5, 3, 36, 24, 74, 9, 12, 75, 34 };
	int n = sizeof(arr) / sizeof(int);

	// print initial array, sort array, then print sorted array
	printArr(arr, n);
	bubbleSort(arr, n);
	printArr(arr, n);
	return 0;
}

// sorts array using the bubble sort algorithm
void bubbleSort(int *arr, int n)
{
	int i, j, temp;

	for (i = 0; i < n - 1; i++)
	{
		for (j = i; j < n - i - 1; j++)
		{
			if (arr[j] > arr[j + 1])
			{
				temp = arr[j];
				arr[j] = arr[j + 1];
				arr[j + 1] = temp;
			}
		}
	}
}

// print an array
void printArr(int *arr, int n)
{
	for (int i = 0; i < n; i++)
	{
		printf("%d ", arr[i]);
	}
	printf("\n");
}

File: C#/Bubble_Sort.cs
using System;

namespace Sort
{
	class Program
	{
		static int Main()
		{
            // initialize array
            int[] arr = { 7, 46, 78, 23, 5, 4, 243, 53, 23 };

            // print original array, sort array, then print sorted array
            printArr(arr);
            bubbleSort(arr);
            printArr(arr);
            return 0;
		}

        // sorts an array using a bubble sort algorithm
        static void bubbleSort(int[] arr)
        {
            // get array's length and initialize variables
            int n = arr.Length;
            int temp;

            // Loop over array n - 1 times. Each iteration sorts one element,
            // except the last iteration, which sorts two.
            for (int i = 0; i < n - 1; i++)
            {
                // compare each pair of adjacent elements in array
                for (int j = 0; j < n - 1 - i; j++)
                {
                    // swap elements if left element is greater than right 
                    // element
                    if (arr[j] > arr[j + 1])
                    {
                        temp = arr[j];
                        arr[j] = arr[j + 1];
                        arr[j + 1] = temp;
                    }
                }
            }
        }

        // prints an array
        static void printArr(int[] arr)
        {
            int n = arr.Length;
            for (int i = 0; i < n; i++)
            {
                Console.Write("{0} ", arr[i]);
            }
            Console.Write("\n");
        }
	}
}